server/src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  BadRequestException,
+  ConflictException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { AuthDto } from './dto/auth.dto';
+import { CreateUserDto } from '../users/dto';
+import { UsersService } from '../users/users.service';
+import { RefreshTokensService } from '../refresh-tokens/refresh-tokens.service';
+import * as bcrypt from 'bcrypt';
+import { authServiceErrorMessages } from './auth.constants';
+import { AuthTokens, JwtPayload } from './types';
+import { JwtService } from '@nestjs/jwt';
+import { AppConfigService } from 'src/config/app-config.service';
+import { UserEntity } from '../users/user.entity';
+import { FindOptionsWhere } from 'typeorm';
+import { v4 as uuidv4 } from 'uuid';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly usersService: UsersService,
+    private readonly refreshTokensService: RefreshTokensService,
+    private readonly jwtService: JwtService,
+    private readonly appConfigService: AppConfigService,
+  ) {}
```

Comment:
move jwt constants to constructor: 

this.jwtAccessSecret = this.appConfigService.get<string>('JWT_ACCESS_SECRET');
etc for other jwt env variables needed